File: riistaandroid/src/main/java/fi/riista/mobile/models/observation/ObservationSpecimen.java
package fi.riista.mobile.models.observation;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;

public class ObservationSpecimen implements Serializable {
    @JsonProperty("id")
    public Long id;

    @JsonProperty("rev")
    public Long rev;

    @JsonProperty("gender")
    public String gender;

    @JsonProperty("age")
    public String age;

    @JsonProperty("state")
    public String state;

    @JsonProperty("marking")
    public String marking;

    @JsonProperty("lengthOfPaw")
    public String lengthOfPaw;

    @JsonProperty("widthOfPaw")
    public String widthOfPaw;
}

File: mobile-common-lib/src/commonMain/kotlin/fi/riista/common/model/ObservationSpecimen.kt
package fi.riista.common.model

import fi.riista.common.dto.ObservationSpecimenDTO
import kotlinx.serialization.Serializable

typealias ObservationSpecimenId = Long

@Serializable
data class ObservationSpecimen(
    val id: ObservationSpecimenId,
    val rev: Int,
    val gender: BackendEnum<Gender>? = null,
    val age: BackendEnum<ObservedGameAge>? = null,
    val state: BackendEnum<ObservedGameState>? = null,
    val marking: BackendEnum<GameMarking>? = null,
    val widthOfPaw: Double? = null,
    val lengthOfPaw: Double? = null,
)

internal fun ObservationSpecimen.toObservationSpecimenDTO(): ObservationSpecimenDTO {
    return ObservationSpecimenDTO(
        id = id,
        rev = rev,
        gender = gender?.rawBackendEnumValue,
        age = age?.rawBackendEnumValue,
        state = state?.rawBackendEnumValue,
        marking = marking?.rawBackendEnumValue,
        widthOfPaw = widthOfPaw,
        lengthOfPaw = lengthOfPaw
    )
}
